conda/common/configuration.py

```diff
@@ -1184,6 +1184,11 @@ def load(self, name, match):
         for attr_name, loaded_child_parameter in loaded_attrs.items():
             object_copy.__setattr__(attr_name, loaded_child_parameter)
 
+        # we do this for the default values which were not copied above
+        for attr_name, parameter_type in vars(self._element_type).items():
+            if attr_name not in value.keys():
+                setattr(object_copy, attr_name, parameter_type.default)
```

Comment:
I agree this fixes it, however I propose we fix this in the above code instead of adding another for-loop (this would replace lines 1169-1180):

```
        object_parameter_attrs = {
            attr_name: parameter_type
            for attr_name, parameter_type in vars(self._element_type).items()
            if isinstance(parameter_type, Parameter)
        }

        # recursively load object fields (loading the default if parameter is undefined)
        loaded_attrs = {}
        for attr_name, parameter_type in object_parameter_attrs.items():
            try:
                raw_child_value = value[attr_name]
            except KeyError:
                loaded_attrs[attr_name] = parameter_type.default
            else:
                loaded_attrs[attr_name] = parameter_type.load(name, value[attr_name])
```

Reply:
Thanks for the suggestion @kenodegard.

I'm not a big fan of `try... except... else...` though, so I came up with something else. Let me know what you think.

Reply:
Nice!!